### PR TITLE
[16.0][IMP] base_edifact: Map PIA segment with product default_code

### PR DESCRIPTION
**Pull request by thienvh332**

### CODE REVIEW

File: base_edifact/models/edifact.py [hunk 1]
```diff
@@ -195,16 +195,25 @@ def map2odoo_product(self, seg, pia=None):
...
+                return {field: code}
         # Fallback on SA if no EAN given
-        if not product[0] and pia[1][0]:
-            return dict(code=pia[1][0])
-        return dict(code=product[0]) if pct == "SRV" else dict(barcode=product[0])
+        elif pia[1][0]:
+            record = product_tmp.search([("default_code", "=", pia[1][0])], limit=1)
+            if record:
+                return {"code": pia[1][0]}
+        return {}
```
Comment (sonhd91) [inline base_edifact/models/edifact.py:216]: Do we have tests case for this ? Incase of the barcode and the code from pia segment not properly provided ?

Comment (thienvh332) [reply base_edifact/models/edifact.py:216]: We have only case is that the barcode declared in LIN does not exist in product and PIA does not exist. Testing again just for PIA doesn't make too much sense.

Comment (thienvh332) [reply base_edifact/models/edifact.py:216]: But i updated it. Could you have a check it again?

**Review by sonhd91 — APPROVED** [review]
LGTM

**Pull request merged**